common/scala/src/main/scala/org/apache/openwhisk/core/database/ActivationStore.scala

```diff
@@ -154,6 +162,23 @@ trait ActivationStore {
     since: Option[Instant] = None,
     upto: Option[Instant] = None,
     context: UserContext)(implicit transid: TransactionId): Future[Either[List[JsObject], List[WhiskActivation]]]
+
+  /**
+   * Check if the system is configured to not store the activation in the database
+   * Dont't store if activation is successful, blocking, not in debug mode and no disable store is configured
+   *
+   * @param isSuccess is successful activation
+   * @param isBlocking is blocking activation
+   * @param debugMode is logging header set to "on" for the invocation
+   * @param disableStore is disable store configured
+   * @return Should the activation be stored to the database
+   */
+  private def shouldStoreActivation(isSuccess: Boolean,
+                                    isBlocking: Boolean,
+                                    debugMode: Boolean,
+                                    disableStore: Boolean): Boolean = {
+    !(isSuccess && isBlocking && !debugMode && disableStore)
```

Comment:
As noted above, I think the alternate with ! and || might be clearer. A nit so defer to you.